test/parallel/test-events-once.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+const common = require('../common');
+const { once, EventEmitter } = require('events');
+const { strictEqual, deepStrictEqual } = require('assert');
+
+async function onceAnEvent() {
+  const ee = new EventEmitter();
+
+  process.nextTick(() => {
+    ee.emit('myevent', 42);
+  });
+
+  const [value] = await once(ee, 'myevent');
+  strictEqual(value, 42);
+}
+
+async function onceAnEventWithTwoArgs() {
+  const ee = new EventEmitter();
+
+  process.nextTick(() => {
+    ee.emit('myevent', 42, 24);
+  });
+
+  const value = await once(ee, 'myevent');
+  deepStrictEqual(value, [42, 24]);
+}
+
+async function catchesErrors() {
+  const ee = new EventEmitter();
+
+  const expected = new Error('kaboom');
+  let err;
+  process.nextTick(() => {
+    ee.emit('error', expected);
+  });
+
+  try {
+    await once(ee, 'myevent');
+  } catch (_e) {
+    err = _e;
+  }
+  strictEqual(err, expected);
+}
+
+async function stopListeningAfterCatchingError() {
+  const ee = new EventEmitter();
+
+  const expected = new Error('kaboom');
+  let err;
+  process.nextTick(() => {
+    ee.emit('error', expected);
+    ee.emit('myevent', 42, 24);
+  });
+
+  process.on('multipleResolves', common.mustNotCall());
+
+  try {
+    await once(ee, 'myevent');
+  } catch (_e) {
+    err = _e;
+  }
```

Comment:
These cases do not verify that `once(ee, 'myevent');` does not reject.

Reply:
But it should reject here, it gets an error emitted.
The first two testcases verify that `once(ee, 'myevent')` doesn't needlessly reject, don't they?
Or are you proposing a testcase for `ee.emit('myevent', 42, 24); ee.emit('error', expected);`?

Reply:
> These cases do not verify that once(ee, 'myevent'); does not reject.

This test verifies that this test rejects with that given error. The actual key of the test is the `'multipleResolves'` assertion anyway.

Reply:
I've added some assertions to verify that no spurious `'error'`  or `'myevent'` listeners are left in place.

Reply:
Yes, I did not check the error assertion down below. I personally would just use `assert.rejects()` as it is clear that this will always verify that the promise rejected. But I am fine as it is.